Dockerfile.build

```diff
@@ -18,7 +19,7 @@ COPY operators/constellation-node-operator/api/go.mod ./operators/constellation-
 COPY operators/constellation-node-operator/api/go.sum ./operators/constellation-node-operator/api/go.sum
 COPY go.mod ./
 COPY go.sum ./
-RUN go mod download all
+RUN --mount=type=secret,id=netrc,dst=/root/.netrc,required=true go mod download all
```

Comment:
This is okay while we keep it away from main but we should maybe add a comment here to ensure this never gets merged by accident.
It should always be possible to build the main and release versions without requiring authentication.